src/Vault.sol

```diff
@@ -1379,45 +1403,41 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, Ownable {
 
   /**
    * @notice Creates `_shares` tokens and assigns them to `_receiver`, increasing the total supply.
-   * @param _receiver Address that will receive the minted shares
-   * @param _shares Shares to mint
    * @dev Emits a {Transfer} event with `from` set to the zero address.
    * @dev `_receiver` cannot be the zero address.
+   * @param _receiver Address that will receive the minted shares
+   * @param _shares Shares to mint
    */
   function _mint(address _receiver, uint256 _shares) internal virtual override {
-    if (_shares > maxMint(_receiver))
-      revert MintMoreThanMax(_receiver, _shares, maxMint(_receiver));
-
-    _twabController.mint(_receiver, SafeCast.toUint96(_shares));
-
+    _twabController.mint(_receiver, SafeCast.toUint112(_shares));
```

Comment:
Here we can rely on the twab controller overflow to prevent excess minting

Reply:
`TwabController.mint` takes a `uint112` argument so we need to cast it if this function takes a uint256 arg.

Reply:
(I meant in terms of the underlying overflow of the balance of the receiver)

Reply:
Exactly. We need to safe cast here first.